x-pack/plugins/security_solution/public/common/components/exceptions/viewer/index.tsx

```diff
@@ -4,96 +4,384 @@
+      color: 'danger',
+      title: i18n.FETCH_LIST_ERROR,
+      iconType: 'alert',
+    }),
+  });
+
+  const setSelectedListType = useCallback(
+    (type: ToggleId): void => {
+      dispatch({
+        type: 'updateSelectedListType',
+        listType: type,
+      });
+    },
+    [dispatch]
+  );
+
+  const setIsModalOpen = useCallback(
+    (isOpen: boolean): void => {
+      dispatch({
+        type: 'updateModalOpen',
+        isOpen,
+      });
+    },
+    [dispatch]
+  );
+
+  const setExceptionToDelete = useCallback(
+    (id: string | null): void => {
+      dispatch({
+        type: 'updateExceptionToDelete',
+        id,
+      });
+    },
+    [dispatch]
+  );
+
+  const onFetchList = useCallback(
+    ({ id, namespaceType }: ApiProps): void => {
+      if (fetchList != null && id != null && namespaceType != null) {
+        fetchList({
+          listId: id,
+          listNamespaceType: namespaceType,
+        });
+      }
+    },
+    [fetchList]
+  );
+
+  const onFiltersChange = useCallback(
+    (filter: { filter: string; tags: string[] }): void => {
+      dispatch({
+        type: 'updateFilterOptions',
+        filterOptions: filter,
+      });
+    },
+    [dispatch]
+  );
+
+  const onAddExceptionItem = useCallback((): void => {
+    setIsModalOpen(true);
+  }, [setIsModalOpen]);
 
+  const onEditExceptionItem = useCallback(
+    (exception: ExceptionListItemSchema): void => {
+      dispatch({
+        type: 'updateExceptionToEdit',
+        exception,
+      });
+
+      setIsModalOpen(true);
+    },
+    [setIsModalOpen]
+  );
+
+  const onCloseExceptionModal = useCallback(
+    ({ actionType, listId, listNamespaceType }): void => {
+      setIsModalOpen(false);
+
+      // TODO: This callback along with fetchList can probably get
+      // passed to the modal for it to call itself maybe
+      if (actionType === ModalAction.CREATE && listId != null && onAssociateList != null) {
+        onAssociateList(listId);
+      }
+
+      onFetchList({ id: listId, namespaceType: listNamespaceType });
+    },
+    [setIsModalOpen, onFetchList, onAssociateList]
+  );
+
+  const onDeleteException = useCallback(
+    async ({ id, namespaceType }: { id: string; namespaceType: NamespaceType }) => {
+      try {
+        const deleteTask = new AbortController();
+        setExceptionToDelete(id);
+        await deleteExceptionListItemById({ http, id, namespaceType, signal: deleteTask.signal });
+        setExceptionToDelete(null);
+
+        onFetchList({ id: selectedListId, namespaceType: selectedListNamespaceType });
+      } catch (error) {
+        onDispatchToaster({
+          color: 'danger',
+          title: i18n.DELETE_EXCEPTION_ERROR,
+          iconType: 'alert',
+        });
+        setExceptionToDelete(null);
+      }
+    },
+    [
+      dispatch,
+      http,
+      setExceptionToDelete,
+      onDispatchToaster,
+      selectedListNamespaceType,
+      selectedListId,
+    ]
+  );
+
+  const getListTypes = ({
```

Comment:
I think it might be cleaner if this function returned a list of `listTypes` only and didn't have the side effect of setting a `selectedListType` as well. It may also be beneficial to wrap it in a `useMemo()`.

Reply:
With the refactor, this code is no longer there.